Torch.Server/ViewModels/ConfigDedicatedViewModel.cs

```diff
@@ -73,12 +75,57 @@ public WorldViewModel SelectedWorld
             }
         }
 
+        public async Task UpdateAllModInfosAsync(Action<string> messageHandler = null)
+        {
+            if (Mods.Count() == 0)
+                return;
+
+            var ids = Mods.Select(m => m.PublishedFileId);
+            var workshopService = WebAPI.Instance;
+            Dictionary<ulong, PublishedItemDetails> modInfos = null;
+
+            try
+            {
+                modInfos = (await workshopService.GetPublishedFileDetails(ids.ToArray()));
+            }
+            catch (Exception e)
+            {
+                Log.Error(e.Message);
+                return;
+            }
+
+            Log.Info($"Mods Info successfully retrieved!");
+
+            foreach (var mod in Mods)
+            {
+                if (!modInfos.ContainsKey(mod.PublishedFileId) || modInfos[mod.PublishedFileId] == null)
+                {
+                    Log.Error($"Failed to retrieve info for mod with workshop id '{mod.PublishedFileId}'!");
+                }
+                //else if (!modInfo.Tags.Contains(""))
+                else
+                {
+                    mod.FriendlyName = modInfos[mod.PublishedFileId].Title;
+                    mod.Description = modInfos[mod.PublishedFileId].Description;
+                    //mod.Name = modInfos[mod.PublishedFileId].FileName;
+                }
+            }
+        }
+
         public List<string> Administrators { get => _config.Administrators; set => SetValue(x => _config.Administrators = x, value); }
 
         public List<ulong> Banned { get => _config.Banned; set => SetValue(x => _config.Banned = x, value); }
         
-        private List<ulong> _mods = new List<ulong>();
-        public List<ulong> Mods { get => _mods; set => SetValue(x => _mods = x, value); }
+        private ObservableCollection<ModItemInfo> _mods = new ObservableCollection<ModItemInfo>();
+        public ObservableCollection<ModItemInfo> Mods
+        {
+            get => _mods;
+            set
+            {
+                SetValue(x => _mods = x, value);
+                Task.Run(() => UpdateAllModInfosAsync());
```

Comment:
Update method will not be called when mods are added or removed, only during initial load. Ensure this is the intended behavior.

Reply:
That's intended. That method only updates the mod metadata from the workshop, so it doesn't need to be run when a mod is removed.
When adding a mod, it's metadata is updated here: https://github.com/TorchAPI/Torch/blob/c44383cf32ab785fc53a5e156337ed287163c391/Torch.Server/Views/ModListControl.xaml.cs#L102

Reply:
But I could also add an observer to the `_mods.CollectionChanged` event. What do you think?

Reply:
If adding checks explicitly, then it's probably fine. Add an event listener if it's easy, but otherwise I wouldn't worry about it.

Reply:
Alright, in that case I'd leave it like this.  If I would add the event listener, I'd either request metadata twice for each added mod or had to implement hacky workarounds to prevent these updates on certain occasions.